src/Confluent.Kafka/Structs/Error.cs

```diff
@@ -64,8 +72,19 @@ public override int GetHashCode()
         public static bool operator !=(Error a, Error b)
             => !(a == b);
 
+        // Returns the string representation of the error.
+        // If a rich error string is available return that, otherwise fall
+        // back to librdkafka's static error code to string conversion.
         public override string ToString()
-            => $"[{(int)Code}] {Message}";
+        {
+            if (!string.IsNullOrEmpty(String))
+                return String;
+            else
+                return ErrorCode2String(Code);
+        }
+
 
+        // Return the human readable representation of a (librdkafka) error code
+        static public string ErrorCode2String(ErrorCode code) => Internal.Util.Marshal.PtrToStringUTF8(Impl.LibRdKafka.err2str(code));
```

Comment:
I'd do this as extension method on `ErrorCode`. I'd put the following in ErrorCode.cs in the Confluent.Kafka namespace:
 
```
    public static class ErrorCodeExtensions
    {
        public static string GetReason(this ErrorCode code)
        {
            return Internal.Util.Marshal.PtrToStringUTF8(Impl.LibRdKafka.err2str(code));
        }
    }
```